From 159b9e2ea427f4c741df8b2465252532ab8f98ff Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Wed, 24 Sep 2014 14:17:29 +0800
Subject: [PATCH] MLK-9606-3 usb: otg: test: remove test maintain timer if HNP
 happens

If B device set host request flag before test maintain timer expires, remove
the timer and clear test maintain session flag, do HNP as a normal OTG device.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit d914c80921652419037d935f14545fe87f74a71c)
---
 drivers/usb/common/usb-otg-fsm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 0cc68b5..870ed7d 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -395,6 +395,11 @@ static int otg_handle_role_switch(struct otg_fsm *fsm, struct usb_device *udev)
 			}
 		}
 		fsm->a_bus_req = 0;
+		if (fsm->tst_maint) {
+			fsm->tst_maint = 0;
+			fsm->otg_vbus_off = 0;
+			otg_del_timer(fsm, A_TST_MAINT);
+		}
 		return HOST_REQUEST_FLAG;
 	} else if (state == OTG_STATE_B_HOST) {
 		fsm->b_bus_req = 0;
-- 
1.8.0

